server/src/QRServer/api/api.py

```diff
@@ -60,19 +65,102 @@ def _add_routes(self):
             web.get('/api/v1/game/stats', self._v1_game_stats),
             web.get('/api/v1/health', self._v1_health),
             web.get('/api/v1/lobby/stats', self._v1_lobby_stats),
+            web.get('/api/v1/tournaments/{id}', self._v1_tournaments),
+            web.get('/api/v1/tournaments/{id}/duels', self._v1_tournament_duels),
+            web.get('/api/v1/tournaments/{id}/matches', self._v1_tournament_matches),
+            web.get('/api/v1/tournaments/{id}/users', self._v1_tournament_users),
         ])
 
-    async def _v1_game_stats(self, _request: web.Request):
+    async def _v1_game_stats(self, _request: web.Request) -> web.Response:
         player_count = self.game_server.get_player_count()
         return web.json_response({
             'player_count': player_count,
         })
 
-    async def _v1_health(self, _request: web.Request):
+    async def _v1_health(self, _request: web.Request) -> web.Response:
         return web.json_response({})
 
     async def _v1_lobby_stats(self, _request: web.Request):
         player_count = self.lobby_server.get_player_count()
         return web.json_response({
             'player_count': player_count,
         })
+
+    async def _v1_tournament_users(self, request) -> web.Response:
+        tournament_id = request.match_info['id']
+        users: list[DbUser] | None = await self.connector.list_tournament_users(tournament_id)
+
+        if users is None:
+            return web.json_response(data=None, status=404)
+
+        users_view: dict[str, list[dict[str, str]]] = {'users': []}
```

Comment:
The type is wrong, it works now but the point of doing `{'users': ... }` is to be extensible. Construct objects in stages instead of deconstructing them just to append something.

This applies to code below too.

Reply:
do you mean to do it like this?
```py
        users_view: list[dict[str, str]] = []

        for user in users:
            user_view = {
                'id': user.user_id,
                'username': user.username,
            }
            users_view.append(user_view)
        return web.json_response(data={'users': users_view}, status=200)
 ```